### ISSUE TITLE
extends causing string literal can't be used as index

### ISSUE THREAD

**Issue by troy351** [microsoft/TypeScript#54886]
# Bug Report

### 🔎 Search Terms

cannot be used as an index type

### 🕗 Version & Regression Information

- This changed between versions 4.9.5 and 5.0.4

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgCLAWZBvAsAKGWRgHsSAuZARgKICM4pKb8BfAgsATwAcUBBZAF40GMAG0A5KRKSAusgD0i5AHcSUANYBnDvm59kAMWHJpZSZ14oAQqfSZxRhcrUadegyiMmR52cgQAB6QIAAm2sgAFP6SyAA+ZgxQkgCUyAD8ZjJxlCAQAG7QSiqgyGAAFhAaXAA0xr5CIkZWhgDC9mJOziXIICRY6lqgAOb1ACrWZgBEMtNxCHAg-Vh0KACu2hBhyHCRS8igYcHl1gB0UQBMAKwAzAAcqUA)

### 💻 Code

```ts
interface Dict {
  foo: 1
  bar: 1
}

type A = Dict['foo'] // works

type F = 'foo'
type B = Dict[F] // works

type FF = 'foo' extends ('foo' | 'bar') ? 'foo' : never // in theory, FF === F
type C = Dict[FF] // not working, Type '"foo"' cannot be used as an index type.(2538)
```

### 🙁 Actual behavior

`type C` throws error

### 🙂 Expected behavior

no error

**Comment by jcalz** [comment]
Could you please edit this to conform to the [bug report template](https://github.com/microsoft/TypeScript/raw/main/.github/ISSUE_TEMPLATE/Bug_report.md)?  I'd hate for this to be closed for not following the template when it looks like a real issue

**Comment by RyanCavanaugh** [comment]
What's the version of this code that you originally had?

**Comment by troy351** [comment]
This is a testing code when I debugging a complicated generic type, and the generic type works. 
I know this kind of code nearly won't happen in real world, but it's still a bug.

**Comment by RyanCavanaugh** [comment]
Right, but I'd really like a "real" version so we can make a testcase in the PR

**Comment by troy351** [comment]
Sorry, there is no "real" version. Maybe consider the provided playground code as real
